option optimize_for = LITE_RUNTIME;
package protbuf;

/// ptype_ prefix add to avoid names conflicts

message ParameterBasicInformation{
	required uint32 id = 1;
}

message Parameter{
  required uint32 id = 1;
  required string name = 2;
  required string symbol=3;

  message Config {
    enum Type{
      INT = 1;
      UINT = 2;
      DOUBLE = 3;

      STRING = 10;
      TEMPERATURE = 20;
      DATE = 30;
    }
    required Type type = 1;
    optional bytes defaultValue = 2;
    optional double minValue = 3;
    optional double maxValue = 4;
    optional int32 minLength = 5;
    optional int32 maxLength = 6;
  }
  required Config config = 4; 
  optional string description=5;
}

message addParameter{
  enum AddMode{
    insert = 1;
    update = 2;
  }
  required AddMode mode = 1;
  required Parameter parameter = 2;
}

message resAddParameter{
  enum addParameterReplay{
    addOk = 1;
    parameterExists = 2;
    updateOk = 3;
    updataFailed_noSuchParameterInDatabase = 4;
  }
  repeated addParameterReplay replay=1;
}

message reqParameters{
  optional bool getDiff = 1;
}

message resParameters{
	repeated Parameter parameters=1;
}
